_posts/2022-10-31-segmentation.md

```diff
@@ -0,0 +1,314 @@
+I will discuss the consequences of this later on when I compare all the methods together.
+
+The main test I wanted to do was to make sure that whatever segmentation this plugin outputs is the same as the Cellpose 2.0 output when given the same exact parameters. 
+I found that there is no difference between the two visually (Figure 7).
+
+{%
+  include figure.html
+  image="images/blog/segmentation/plugin_vs_cellpose.png"
+  width="100%"
+%}
+
+> Figure 7. Comparison between the CellProfiler Cellpose plugin and Cellpose 2.0. This figure demonstrates when using the same parameters for both methods, the segmentation is the exact same visually.
+
+This means that I am fully confident in being able to replicate the same segmentations that I got when running a pipeline with Cellpose 2.0 in a pipeline that uses the CellProfiler Cellpose plugin as the segmenting method.
+
+## Proposed Improvements for CellProfiler Cellpose Plugin
+
+One thing that I noticed right off the bat is that this module did not have (but **needed!**) is a way to remove objects/cells/nuclei that had pixels that touched an edge of the image. 
+This is a function that is already in place in the CellProfiler standard method and in the [Cellpose pipeline](https://github.com/jenna-tomkinson/NF1_SchwannCell_data/blob/548983b6cd32ed4121e719e34eb84934185ae0c6/2_segmenting_data/segmentation_utils.py#L37-L39).
+I use for all my CellProfiler projects, and I thought it would be a good idea to add it to the CellProfiler Cellpose plugin!
+
+As you can see in [the PR](https://github.com/CellProfiler/CellProfiler-plugins/pull/169) I made for CellProfiler, I was able to take one of the utility functions already in Cellpose, which removes objects that touch the edges, and added it to the module to allow the option for users.
+
+After all of this, I believe that this plugin is a great option for Cellprofiler users to have the ability to choose what type of segmentation method they want to use. 
+I am excited to be able to use this and to help with the improvement of the module!
+
+## Comparison of the three methods
+
+One of the biggest differences I noticed between Cellpose and CellProfiler was that Cellpose segmentations did not fit the shape of the cells as much as CellProfiler. 
+As well, CellProfiler included more of the stringy parts of the actin that Cellpose did not (Figure 8).
+
+{%
+  include figure.html
+  image="images/blog/segmentation/sensitivity_blog_fig.png"
+  width="100%"
+%}
+
+> Figure 8. Comparison of sensitivity between Cellpose and CellProfiler. This figure shows how CellProfiler segments cells much tighter and includes small parts compared to Cellpose. Cellpose segments cells more broadly and includes more of the background compared to CellProfiler.
+
+After doing a lot of prototyping with both CellProfiler and Cellpose, I believe that Cellpose segments whole cells much better (e.g. no cell is segmented over each other) than CellProfiler (Figure 9).
```

Comment:
wait a sec, I thought you just said in lines 261 and 262:

> One of the biggest differences I noticed between Cellpose and CellProfiler was that Cellpose segmentations did not fit the shape of the cells as much as CellProfiler. 
> As well, CellProfiler included more of the stringy parts of the actin that Cellpose did not (Figure 8).

Doesn't this contradict that Cellpose segments whole cells much better? Where am I confused?

Reply:
@gwaybio 

From my perspective, I am comparing two different metrics; sensitivity and whole cell segmentation. 

CellProfiler is much more sensitive, but Cellpose segments the whole cell more accurately/better.

Reply:
got it!